src/VisualStudio/Core/Def/Implementation/Diagnostics/FxCopAnalyzersSuggestedActionCallback.cs

```diff
@@ -139,16 +146,27 @@ private bool IsNuGetInstalled()
                     return false;
                 }
 
+                var foundAnalyzerReference = false;
                 foreach (var analyzerReference in document.Project.AnalyzerReferences)
                 {
-                    if (ChildAnalyzerNuGetPackageIds.Contains(analyzerReference.Display))
+                    if (analyzerReference is UnresolvedAnalyzerReference)
+                    {
+                        hasUnresolvedAnalyzerReference = true;
+                        continue;
+                    }
+
+                    var display = analyzerReference.Display;
+                    if (display != null &&
+                        ChildAnalyzerNuGetPackageIds.Contains(display))
```

Comment:
So the display string here is actually being used for checks? That seems....dangerous. What form is the display expected to keep here? Can we add some tests to ensure the form stays in the shape we want?

Reply:
If the display string changes, then we will start getting false positives from this component - it will suggest you to install the package even if you already have it. 

> Can we add some tests to ensure the form stays in the shape we want?
Sure, I will add some tests.

Reply:
@jasonmalinowski  From the original issue #39092:
`_underlying.Display` contains _[ed: contained, at the point of the crash]_ the string
`Unresolved: MyProjPath\\mscorlib.dll`, where MyProjPath is the local filesystem absolute (with doubled-up backslashes) path to the directory holding the .csproj.

One could question the silliness of checking such a name in this context (not to mention I find it somewhat less likely to find analyzers in mscorlib.dll, even _if_ a copy was found alongside the .csproj), but considering it's Monty Python's 50th anniversary... :-)

Reply:
@jasonmalinowski Added regression test with https://github.com/dotnet/roslyn/pull/39097/commits/baf4026a401342cec9e01ca1354d03ca741c6319